test/impl-base/src/main/java/org/jboss/arquillian/test/impl/TestContextHandler.java

```diff
@@ -59,7 +60,7 @@ public class TestContextHandler {
 
     // Since there can be multiple AfterTestLifecycleEvents (After/AfterRules)
     // and we don't know which is the last one, perform the clean up in AfterClass.
-    private Map<Class<?>, Set<Object>> activatedTestContexts = new HashMap<Class<?>, Set<Object>>();
+    private Map<Class<?>, Set<Object>> activatedTestContexts = new ConcurrentHashMap<>();
```

Comment:
This seems to already be guarded by itself. I don't think we need to change the type.